src/Http/Middleware/IdpMiddleware.php

```diff
@@ -69,6 +66,11 @@ public function handle(Request $request, Closure $next)
                 $this->addExtraParametersToUser($user);
 
                 Auth::setUser($user);
+
+                if ($request->query('token')) {
+                    $request->query->remove('token');
+                    return redirect($request->fullUrl());
```

Comment:
Così va in loop perchè `fullUrl` ignora il `remove` della linea precedente.
Puoi fare```                    return redirect(trim($request->url() . "?" . http_build_query($request->query->all()), "?"));
```
non bellissimo ma funziona